Add HTTP header tag explicitly allowing local network access

bogdanni opened the pull request:
Chrome has recently started to issue a warning for the access of the SAMP hub HTTP from HTTPS web pages. This addition silences the warning. 

See https://developer.chrome.com/blog/private-network-access-update

bogdanni committed:
Add HTTP header tag explicitly allowing local network access

See https://developer.chrome.com/blog/private-network-access-update

mbtaylor commented:
@bogdanni thanks for picking this up.  I will probably accept the PR and make a new release with this change, but I want to understand the details some more first.  I have not so far managed to reproduce the issue.  I've upgraded Chrome (on Linux) to a version that identifies itself as `122.0.6261.111 (Official Build)`, which I think means Chrome 122, but it seems to support Web SAMP (JSAMP 1.3.7) from HTTPS (e.g. https://vizier.unistra.fr/viz-bin/VizieR-4) fine without any warning or problem that I can see.  The documentation at the URL you gave suggests this change will be complete at Chrome 117.  What Chrome versions have you seen Web SAMP problems at?

bogdanni commented:
I'm seeing this on macOS 14.4 in Chrome 122.0.6261.112 (Official Build) (arm64).

Here the screenshot from the Inspector with the warning:
![image](https://github.com/mbtaylor/jsamp/assets/1223825/f02d5cdd-734d-4f33-b67d-73b720ce47d2)

This is with the combination between https://www.sidc.be/EUI/data/states/ and the program from https://www.jhelioviewer.org which runs a JSAMP 1.3.7 hub.

bogdanni commented:
Here are more links about the topic:
https://developer.chrome.com/blog/private-network-access-preflight
https://bugs.webkit.org/show_bug.cgi?id=171934

The various dates and versions for which those restrictions were to be imposed in Chrome were obviously not respected...

mbtaylor committed:
Slight changes to Private Network Access implementation

Following the recommendations in
https://wicg.github.io/private-network-access/,
set the Access-Control-Allow-Private-Network key only in response
to the Access-Control_Request-Private-Network key.
Also add a little bit more explanation in the comments.

mbtaylor commented:
OK thanks I see that warning now.  I made some minor tweaks to the implementation and comments.  Does it still look OK to you?

bogdanni commented:
Works great and without warnings, thanks! This is most likely the right implementation.

Btw, had errors from my `maven`:
```
[ERROR] Source option 1.4 is no longer supported. Use 8 or later.
[ERROR] Target option 1.4 is no longer supported. Use 8 or later.
```
Had to change all `1.4` to `8`.

mbtaylor committed:
Minor clarifications in comments

mbtaylor commented:
I have commented on this point on the apps@ivoa.net mailing list: http://mail.ivoa.net/pipermail/apps/2024-March/001667.html

mbtaylor commented:
I have made a new JSAMP release v1.3.8 including this change (release announced [here](http://mail.ivoa.net/pipermail/apps/2024-March/001668.html)).
